### PR TITLE
Use embedder.tokenize in data_sources/tokenize

### LINKED ISSUE #5296: Fix embedder usage in `data_sources_tokenize`

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes: https://github.com/dust-tt/dust/issues/5296
Also fixes context_size for openai/azure_openai. Had no impact since only used for "generate" which is quite legacy

Will pick the right tokenizer for data sources (cl100k) instead of falling back to (cl50k which is inaccurate).

## Risk

None

## Deploy Plan

- deploy core

### CODE REVIEW

File: core/src/providers/openai.rs [hunk 1]
```diff
@@ -1440,24 +1440,32 @@ impl LLM for OpenAILLM {
...
-        }
-        if self.id.starts_with("gpt-4-32k") {
-            return 32768;
-        }
-        if self.id.starts_with("gpt-4-1106-preview") {
-            return 128000;
+        // Reference: https://platform.openai.com/docs/models
+
+        // gpt-3.5-*
+        if self.id.starts_with("gpt-3.5") {
```
Comment (flvndvd) [inline core/src/providers/openai.rs:1446]: Duplicated code, we can call this function from the Azure code, right?

**Review by flvndvd — APPROVED** [review]
LGTM, left one suggestion

File: core/src/providers/openai.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/openai.rs:1446]: Sure I'll create a function :p

Comment (spolu) [reply core/src/providers/openai.rs]: (done)

**Pull request merged**